src/utilities/images.js

```diff
@@ -0,0 +1,32 @@
+export const loadImageSizes = (files) => {
+  return Promise.all( files.map( file => {
+    return new Promise((resolve, reject) => {
+
+      // Check if the file is not an image #fail-fast
+      if( !file.uploadTaskSnaphot.metadata.contentType.startsWith('image')) {
+        resolve(file);
+      }
+
+      // Create a new Image element
+      const img = new Image();
+
+      // Set the onload callback
+      img.onload = event => {
+        // Get image dimensions
+        const { width, height } = event.target;
+
+        // Save dimensions in the file item
+        file.width = width;
+        file.height = height;
+
+        // resolve and return the file with the dimensions
+        resolve(file);
+      };
+
+      img.onerror = () => resolve({});
```

Comment:
have you been able to test this fail?

Reply:
If you make a fake image file that gets accepted this never happens as the download url will already fail:

```
TypeError: Cannot read property 'downloadURL' of undefined
    at UploadField.jsx:46
    at Array.map (<anonymous>)
    at UploadField.jsx:46
    at <anonymous>
```

Reply:
Because this returns a blank file on error the chain continues and then in the next `.then(` in UploadField errors with an array with just one element which is an empty object. I'm guessing this resolve needs to be a more explicit error or the next chained function needs to check the values passed.

Reply:
maybe it should return nothing, like exclude the file. or return the file without dimensions, or throw an error